### ISSUE TITLE
Odd error all of a sudden - ValueError: not enough values to unpack (expected 7, got 1)

### ISSUE THREAD

**Issue by johntdyer** [matt8707/youtube-watching#12]
```
ERROR:waitress:Exception when servicing <waitress.channel.HTTPChannel connected 127.0.0.1:58430 at 0x7fd1ad035750>
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/http/cookiejar.py", line 2041, in _really_load
    domain, domain_specified, path, secure, expires, name, value = \
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: not enough values to unpack (expected 7, got 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/youtube-watching/./app/main.py", line 26, in yt_history
    cookie_jar.load(ignore_discard=True, ignore_expires=True)
  File "/usr/local/lib/python3.11/http/cookiejar.py", line 1802, in load
    self._really_load(f, filename, ignore_discard, ignore_expires)
  File "/usr/local/lib/python3.11/http/cookiejar.py", line 2081, in _really_load
    raise LoadError("invalid Netscape format cookies file %r: %r" %
http.cookiejar.LoadError: invalid Netscape format cookies file '/youtube-watching/config/youtube.com_cookies.txt': 'HqO3kR8-SWu-FcVeoymJ6c0nk0iqHk'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/waitress/task.py", line 84, in handler_thread
    task.service()
  File "/usr/local/lib/python3.11/site-packages/waitress/channel.py", line 428, in service
    task.service()
  File "/usr/local/lib/python3.11/site-packages/waitress/task.py", line 168, in service
    self.execute()
  File "/usr/local/lib/python3.11/site-packages/waitress/task.py", line 434, in execute
    app_iter = self.channel.server.application(environ, start_response)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 2548, in __call__
    return self.wsgi_app(environ, start_response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask_restful/__init__.py", line 467, in wrapper
    resp = resource(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/views.py", line 107, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask_restful/__init__.py", line 582, in dispatch_request
    resp = meth(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/youtube-watching/./app/main.py", line 100, in get
    return yt_history(os.environ['COOKIE'])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/youtube-watching/./app/main.py", line 29, in yt_history
    sys.exit()
SystemExit

```

docker-compose.yaml

```
  youtube-watching:
    container_name: youtube-watching
    image: ghcr.io/matt8707/youtube-watching
    volumes:
      - /etc/youtube-watching/config/:/youtube-watching/config/
    environment:
      - COOKIE=/youtube-watching/config/youtube.com_cookies.txt
    network_mode: bridge
    ports:
      - 5678:5678
    restart: always
    labels:
      - wud.display.name=Youtube Watcher
      - wud.display.icon=mdi:youtube
      - wud.watch=true
      - wud.tag.include=^\d\d\d\d+\.\d+$$

      - homepage.group=Media
      - homepage.name=Youtube Watcher
      - homepage.description=Media Service
      - homepage.icon=youtube.png
    healthcheck:
      test: wget --no-verbose --tries=1 --no-check-certificate --spider http://localhost:5678
      interval: 60s
      timeout: 15s
      retries: 2
      start_period: 10s
```

redacted cookie file 

```
# Netscape HTTP Cookie File
# https://curl.haxx.se/rfc/cookie_spec.html
# This is a generated file! Do not edit.

#HttpOnly_.youtube.com	TRUE	/	TRUE	1729281244	LOGIN_INFO	xxxxxxxxxx-xxxxxx:xxxxxxxx
#HttpOnly_.youtube.com	TRUE	/	TRUE	1696702389	NID	511=xxxx-xxx-xxxxxxxx-xxx
#HttpOnly_.youtube.com	TRUE	/	FALSE	1755986005	HSID	xxxxxxxxxxxxxxx
#HttpOnly_.youtube.com	TRUE	/	TRUE	1755986005	SSID	xxxxx
.youtube.com	TRUE	/	FALSE	1755986005	APISID	xxxxxxxxx-/AIxAS-xxxxxxx
.youtube.com	TRUE	/	TRUE	1755986005	SAPISID	xxxxxxxxxx/xxxxxxxxx
.youtube.com	TRUE	/	TRUE	1755986005	__Secure-1PAPISID	xxxxxxxxxx/xxxx
.youtube.com	TRUE	/	TRUE	1755986005	__Secure-3PAPISID	xxxx/xxxxxxx
#HttpOnly_.youtube.com	TRUE	/	TRUE	1708466005	VISITOR_PRIVACY_METADATA	xxxxxxxx%3D
.youtube.com	TRUE	/	FALSE	1755986005	SID	xxxxxxxxx-Q_ZOLb-ssssssss.
#HttpOnly_.youtube.com	TRUE	/	TRUE	1755986005	__Secure-1PSID	xxxxxxxxxxxxxxxxx-Q_ZOLb-xxxxxxx-JmYULtw.
#HttpOnly_.youtube.com	TRUE	/	TRUE	1755986005	__Secure-3PSID	xxxxxxxxxxxxxxxxx-Q_ZOLb-xxxxxx-s4q4flPytlpaw.
#HttpOnly_.youtube.com	TRUE	/	TRUE	0	YSC	xxxxxxxxxxxxx
.youtube.com	TRUE	/	FALSE	0	wide	1
#HttpOnly_.youtube.com	TRUE	/	TRUE	1709944159	VISITOR_INFO1_LIVE	xxxxxxxxxxxxxxxxx
#HttpOnly_.youtube.com	TRUE	/	TRUE	1725972447	__Secure-1PSIDTS	sidts-xxxxx
#HttpOnly_.youtube.com	TRUE	/	TRUE	1725972447	__Secure-3PSIDTS	sidts-xxxxx
.youtube.com	TRUE	/	TRUE	1757508756	PREF	f6=80&tz=America.New_York&f4=xxxxxx&f5=20000
.youtube.com	TRUE	/	FALSE	1725972767	SIDCC	xxxxxxxxxxxxxxx-xxxx-BEVXX
#HttpOnly_.youtube.com	TRUE	/	TRUE	1725972767	__Secure-1PSIDCC	xxxxx-alWgVGUHh4nZ1cQwX8cLlsDelXf3T9dI0NjLkwiZ5S32iQ3IqaJjVPG
#HttpOnly_.youtube.com	TRUE	/	TRUE	1725972767	__Secure-3PSIDCC	xxxxx-SWu-xxxxxxxxxx

```

**Comment by matt8707** [comment]
ERROR ”invalid Netscape format cookies file”, try generating a new one without editing it

**Comment by johntdyer** [comment]
So now I've seen something really strange... I tried a incognito tab, logged in and exported.  Restartted the container and it worked for 2 requests.  Then 30 mins later I went back to see if it was still working and it crashed w/ the same invalid cookie error..  The container reports being up for 54 mins so I know it didnt restart ./....

**Comment by matt8707** [comment]
try latest, it'll log any cookie_path errors

```bash
docker pull ghcr.io/matt8707/youtube-watching:2023.9
```

**Comment by johntdyer** [comment]
Worked for about 60 seconds this time... 

root@debian:/mnt/pve/proxmox_nfs/shared-configs# docker inspect youtube-watching | jq '.[].Image'
"sha256:03f8da855303db58ab2b21a5282c4562b31dd509fe24213c5acf569e77accbc0"



```
^C
root@debian:/mnt/pve/proxmox_nfs/shared-configs# docker logs -n 1000000 youtube-watching
ERROR:main:Exception on / [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask_restful/__init__.py", line 489, in wrapper
    resp = resource(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/views.py", line 109, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask_restful/__init__.py", line 604, in dispatch_request
    resp = meth(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/youtube-watching/./app/main.py", line 108, in get
    return yt_history(os.environ['COOKIE'])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/youtube-watching/./app/main.py", line 87, in yt_history
    key = path[0]["videoRenderer"]
          ~~~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'videoRenderer'
ERROR:main:Exception on / [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask_restful/__init__.py", line 489, in wrapper
    resp = resource(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/views.py", line 109, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask_restful/__init__.py", line 604, in dispatch_request
    resp = meth(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/youtube-watching/./app/main.py", line 108, in get
    return yt_history(os.environ['COOKIE'])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/youtube-watching/./app/main.py", line 87, in yt_history
    key = path[0]["videoRenderer"]
          ~~~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'videoRenderer'
ERROR:main:Exception on / [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask_restful/__init__.py", line 489, in wrapper
    resp = resource(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/views.py", line 109, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask_restful/__init__.py", line 604, in dispatch_request
    resp = meth(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/youtube-watching/./app/main.py", line 108, in get
    return yt_history(os.environ['COOKIE'])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/youtube-watching/./app/main.py", line 87, in yt_history
    key = path[0]["videoRenderer"]
          ~~~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'videoRenderer'
ERROR:main:Exception on / [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask_restful/__init__.py", line 489, in wrapper
    resp = resource(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/views.py", line 109, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask_restful/__init__.py", line 604, in dispatch_request
    resp = meth(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/youtube-watching/./app/main.py", line 108, in get
    return yt_history(os.environ['COOKIE'])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/youtube-watching/./app/main.py", line 87, in yt_history
    key = path[0]["videoRenderer"]
          ~~~~~~~^^^^^^^^^^^^^^^^^
KeyError: 'videoRenderer'
root@debian:/mnt/pve/proxmox_nfs/shared-configs#
```

**Comment by matt8707** [comment]
added more logging

```bash
docker pull ghcr.io/matt8707/youtube-watching:2023.9.1
```

**Comment by johntdyer** [comment]
So basically no logging at all now until I exit the container, and even then its just stdout 

```
youtube-watcher --env COOKIE="/youtube-watching/config/newtab_cookies2.txt" -v "/etc/youtube-watching/config:/youtube-watching/config" ghcr.io/matt8707/youtube-watching:2023.9.1
^CFailed to extract video details from YouTube data: 'videoRenderer'
Failed to extract video details from YouTube data: 'videoRenderer'
Failed to extract video details from YouTube data: 'videoRenderer'
Failed to extract video details from YouTube data: 'videoRenderer'
Failed to extract video details from YouTube data: 'videoRenderer'
Failed to extract video details from YouTube data: 'videoRenderer'
Failed to extract video details from YouTube data: 'videoRenderer'
root@debian:/mnt/pve/proxmox_nfs/shared-configs#
```

I even got a new cookie, which almost always works for a bit, and I get a good api response 

```
deo_id": "5mp2vCZ2iFc", "duration_string": "4:27", "thumbnail": "https://img.youtube.com/vi/5mp2vCZ2iFc/maxresdefault.jpg", "original_url": "https://www.youtube.com/watch?v=5mp2vCZ2iFc"}
* Connection #0 to host 192.168.100.3 left intact
root@debian:~# curl http://192.168.100.3:5679/
{"channel": "Ricky Impey", "title": "Simple Temperature Tower In Cura 5.1.0 - It's Now SO Easy!", "video_id": "5mp2vCZ2iFc", "duration_string": "4:27", "thumbnail": "https://img.youtube.com/vi/5mp2vCZ2iFc/maxresdefault.jpg", "original_url": "https://www.youtube.com/watch?v=5mp2vCZ2iFc"}
root@debian:~#
```

but I get no logging when running the container until I exit it 

TLDR; I dont think the logging is working as intended

**Comment by matt8707** [comment]
yeah, it's not proper docker logging you have to bind your terminal to the container, or just `curl http://192.168.100.3:5679` to see the error there

```
Failed to extract video details from YouTube data: 'videoRenderer'
```
it seems like you actually have the data but something goes wrong here, I've added more info on error

```bash
docker pull ghcr.io/matt8707/youtube-watching:2023.9.2
```

**Comment by johntdyer** [comment]
so I have zero clue why based on the change log but 2023.9.2 seems to be working for over 24 hours now....  If you dont mint I would love to keep this open for a few more days just incase it happens again?

**Comment by johntdyer** [comment]
Finally expired but I am not sure what exactly you're looking for in the logs
[youtube.log](https://github.com/matt8707/youtube-watching/files/12676991/youtube.log)

**Comment by johntdyer** [comment]
Note, I did have a internet outage a few hours ago, but my IP didnt change or anything after coming back up

**Comment by matt8707** [comment]
I'm just looking for what goes wrong and it's like you said, cookie expires. There's nothing I can do if google is kicking you out

**Comment by johntdyer** [comment]
ok, but the cookie still seems to work w/ youtube-dl which is why I thought it was application related

**Comment by matt8707** [comment]
Right, but "seems to work", have you actually tested it? Cookies in youtube-dl are for age restricted videos and such

**Comment by johntdyer** [comment]
not sure I follow you're last statement but here is what I tried...

Cookie on my laptop 

```
░▒▓    ~/Des/temp ▓▒░ youtube-dl --cookies ~/Desktop/cookies.txt  "https://www.youtube.com/watch?v=hIk79WmVg4o"
[youtube] hIk79WmVg4o: Downloading webpage
[dashsegments] Total fragments: 14
[download] Destination: FOUNDATION Season 2 Ending Explained!-hIk79WmVg4o.f137.mp4
[download]  34.3% of ~140.00MiB at 13.72MiB/s ETA 00:06

```

Same cookie in container 

```
  youtube-watching:
    container_name: youtube-watching
    image: ghcr.io/matt8707/youtube-watching:2023.9.2
    volumes:
      - /etc/youtube-watching/config/:/youtube-watching/config/
    environment:
      - COOKIE=/youtube-watching/config/newtab_cookies2.txt
    network_mode: bridge
    ports:
      - 5678:5678
    restart: always
    labels:
      - wud.display.name=Youtube Watcher
      - wud.display.icon=mdi:youtube
      - wud.watch=true
      - wud.tag.include=^\d\d\d\d+\.\d+$$
      # - wud.tag.transform=^(\d\d\d\d\.\d+).*$$ => $$1
      - homepage.group=Media
      - homepage.name=Youtube Watcher
      - homepage.description=Media Service
      - homepage.icon=youtube.png
      - ```

```

```
root@debian:/mnt/pve/proxmox_nfs/shared-configs# sha256sum /etc/youtube-watching/config/newtab_cookies2.txt
a62530f5d9d7f92e23e9f9a6f699e02d3db7248e9f2763c1dc0c4d4b4c61a3ee  /etc/youtube-watching/config/newtab_cookies2.txt
root@debian:/mnt/pve/proxmox_nfs/shared-configs#
```


How else should I be testing it ?

**Comment by johntdyer** [comment]
btw, ty again for even taking the time to help me w/ this... it is appreciated

**Comment by matt8707** [comment]
what i mean is that https://www.youtube.com/watch?v=hIk79WmVg4o is a public video, youtube-dl ignores your expired cookie. cookies are only needed when you download private/ age restricted videos

what youtube-watching is doing is using your session to get latest video in your watch history, therefore cookie is always needed

**Comment by johntdyer** [comment]
ah.... ok... so that makes more sense... thank you for explaining it that way...  So any ideas why my cookie expires so much faster then any else's?

**Comment by matt8707** [comment]
I don't

**Comment by hai-nd** [comment]
I also have the same error:
    Failed to extract video details from YouTube data: 'videoRenderer'
I can use youtube-dl to download Youtube files in Debian terminal, but I can not get the {{ value_json.thumbnail }}. I have tried using with youtube-waching docker container and home assistant without youtube-waching docker way, but I can not success...
I am looking forward to have good information from you. Thank you very much!
